src/thingsboard.c

```diff
@@ -189,6 +196,84 @@ static void client_request_time(struct k_work *work)
 	k_work_reschedule(k_work_delayable_from_work(work), K_SECONDS(10));
 }
 
+static int client_handle_rpc_response(struct coap_client_request *req, struct coap_packet *response)
+{
+	const uint8_t *payload;
+	uint16_t payload_len;
+	uint8_t code;
+	char code_str[5];
+	char expected_code_str[5];
+
+	LOG_INF("%s", __func__);
+
+	code = coap_header_get_code(response);
+	if (code != COAP_RESPONSE_CODE_CONTENT) {
+		coap_response_code_to_str(code, code_str);
+		coap_response_code_to_str(COAP_RESPONSE_CODE_CONTENT, expected_code_str);
+		LOG_ERR("Unexpected response code for RPC request: got %s, expected %s", code_str,
+			expected_code_str);
+		return -1;
+	}
+
+	payload = coap_packet_get_payload(response, &payload_len);
+	if (!payload_len) {
+		LOG_ERR("Received an empty RCP response");
+		return payload_len;
```

Comment:
This path does not give the semaphore. Maybe work with a goto here, because you want to give the semaphore either way.

Reply:
good call, I will add it